src/input/mod.rs

```diff
@@ -152,22 +150,30 @@ impl<'a> InputFile<'a> {
 
 
     /// Reads image data into the given FrameBufferMut.
+    ///
+    /// The passed FrameBufferMut must match the image's resolution exactly, and
+    /// the complete image will be read into the FrameBufferMut.
     pub fn read_pixels(&mut self, framebuffer: &mut FrameBufferMut) -> Result<()> {
         // ^^^ NOTE: it's not obvious, but this does indeed need to take self as
         // &mut to be safe.  Even though it is not conceptually modifying the
         // thing (typically a file) that it's reading from, it still has a
         // cursor getting incremented etc. during reads, so the reference needs
         // to be unique to avoid unsafe aliasing.
-        let w = self.header().data_window();
-        if (w.max.x - w.min.x) as usize != framebuffer.dimensions().0 - 1 ||
-           (w.max.y - w.min.y) as usize != framebuffer.dimensions().1 - 1 {
-            panic!("framebuffer size {}x{} does not match input file dimensions {}x{}",
-                   framebuffer.dimensions().0,
-                   framebuffer.dimensions().1,
-                   w.max.x - w.min.x,
-                   w.max.y - w.min.y)
+
+        // Validation
+        if self.header().data_dimensions().0 != framebuffer.dimensions().0 ||
+           self.header().data_dimensions().1 != framebuffer.dimensions().1 {
```

Comment:
Couldn't this be written `self.header().data_dimensions() != framebuffer.dimensions()`?